[DISCORD:1418056806684561568] The latest version cannot install private packages.

This issue was created from Discord post: https://discord.com/channels/1309558646228779139/1418056806684561568

When running mastra build, the following error occurs:

```sh
ackend:build: INFO [2025-09-18 10:46:23.071 +0900] (Mastra CLI): ERR_PNPM_FETCH_404  GET https://registry.npmjs.org/@my-org%2Fprivate-package: Not Found - 404
backend:build: 
backend:build: This error happened while installing the dependencies of backend
backend:build: 
backend:build: @my-org/private-package is not in the npm registry, or you have no permission to fetch it.
backend:build: 
backend:build: No authorization header was set for the request.
```
Version in which the problem occurred
```
  '@mastra/ai-sdk': 0.0.3
  '@mastra/client-js': 0.13.0
  '@mastra/core': 0.17.0
  '@mastra/evals': 0.13.7
  '@mastra/mcp': 0.13.0
  '@mastra/memory': 0.15.2
  '@mastra/mongodb': 0.14.1
  'mastra': 0.13.0
```

Version without issues
```
  '@mastra/ai-sdk': 0.0.2
  '@mastra/client-js': 0.12.2
  '@mastra/core': 0.16.2
  '@mastra/evals': 0.13.5
  '@mastra/mcp': 0.12.0
  '@mastra/memory': 0.15.0
  '@mastra/mongodb': 0.14.0
  mastra: 0.12.2
```

--- a/.changeset/late-poems-type.md
+++ b/.changeset/late-poems-type.md
@@ -1,0 +1,5 @@
+---
+'@mastra/deployer': patch
+---
+
+build bundle - Configure project root for private packages config

--- a/packages/deployer/src/bundler/index.ts
+++ b/packages/deployer/src/bundler/index.ts
@@ -465,12 +465,12 @@ export const tools = [${toolsExports.join(', ')}]`,
       this.logger.info('Done copying public files');
 
       this.logger.info('Copying .npmrc file');
-      await this.copyDOTNPMRC({ outputDirectory });
+      await this.copyDOTNPMRC({ outputDirectory, rootDir: projectRoot });
 
       this.logger.info('Done copying .npmrc file');
 
       this.logger.info('Installing dependencies');
-      await this.installDependencies(outputDirectory);
+      await this.installDependencies(outputDirectory, projectRoot);
 
       this.logger.info('Done installing dependencies');
     } catch (error) {

